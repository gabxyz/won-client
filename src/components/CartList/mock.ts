export default [
  {
    id: '1',
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    title: 'Red Dead Redemption 2',
    basePrice: null,
    price: '$25.00'
  },
  {
    id: '2',
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    title: 'Borderlands 3',
    basePrice: null,
    price: '$40.00'
  }
]
