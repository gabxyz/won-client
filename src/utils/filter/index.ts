import { ItemProps } from 'components/ExploreSidebar'
import { ParsedUrlQueryInput } from 'querystring'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
  filterItems: Pick<ItemProps, 'type' | 'name'>[]
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const parsePrice = (price: string, obj: any) => {
  if (price[0] === '0' || price[0] === 'u') {
    obj['price_lte'] =
      price.length < 1 ? Number(price[0]) : Number(price.slice(1))
  }

  if (price[0] === 'a') {
    obj['price_gte'] = Number(price.slice(1))
  }
}

export const parseQueryStringToWhere = ({
  queryString,
  filterItems
}: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  Object.keys(queryString)
    .filter((item) => item !== 'sort')
    .forEach((key) => {
      const item = filterItems?.find((item) => item.name === key)
      const isCheckbox = item?.type === 'checkbox'

      item?.name === 'price'
        ? parsePrice(queryString[key]!.toString(), obj)
        : (obj[key] = !isCheckbox
            ? queryString[key]
            : { name_contains: queryString[key] })
    })

  return obj
}

export const parseQueryStringToFilter = ({
  queryString,
  filterItems
}: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  Object.keys(queryString).forEach((key) => {
    const item = filterItems?.find((item) => item.name === key)
    const isCheckbox = item?.type === 'checkbox'
    const isArray = Array.isArray(queryString[key])

    obj[key] = !isArray && isCheckbox ? [queryString[key]] : queryString[key]
  })

  return obj
}
