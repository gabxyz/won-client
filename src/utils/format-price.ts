export default function formatPrice(price: number | bigint): string {
  if (price === 0) {
    return 'FREE'
  }
  if (price === null) {
    return 'Coming Soon'
  }
  return new Intl.NumberFormat('en', {
    style: 'currency',
    currency: 'USD'
  }).format(price)
}
