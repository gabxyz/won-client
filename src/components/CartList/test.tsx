import { render, screen } from 'utils/test-utils'

import CartList from '.'
import mockItems from './mock'

describe('<CartList />', () => {
  it('should render the cart list', () => {
    const { container } = render(<CartList items={mockItems} total="$65.00" />)

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText('$65.00')).toHaveStyle({ color: '#F231A5' })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button', () => {
    render(<CartList items={mockItems} hasButton />)
    expect(screen.getByText(/buy it now/i)).toBeInTheDocument()
  })

  it('should render empty if there are no games', () => {
    render(<CartList />)

    expect(screen.getByText(/your cart is empty/i)).toBeInTheDocument()
    expect(screen.queryByText(/total/i)).not.toBeInTheDocument()
  })
})
