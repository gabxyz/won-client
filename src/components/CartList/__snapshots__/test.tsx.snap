// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the cart list 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

.c2 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c4 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c8 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  gap: 0.8rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c1 {
  max-height: 40rem;
  overflow-y: auto;
}

.c10 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c4 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Red Dead Redemption 2"
            src="https://source.unsplash.com/user/willianjusten/151x70"
          />
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Red Dead Redemption 2
          </h3>
          <div
            class="c7 c8"
            role="price"
          >
            <div
              class="c9"
            >
              $25.00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Borderlands 3"
            src="https://source.unsplash.com/user/willianjusten/151x70"
          />
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Borderlands 3
          </h3>
          <div
            class="c7 c8"
            role="price"
          >
            <div
              class="c9"
            >
              $40.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c10"
  >
    <span>
      Total:
    </span>
    <span
      class="c11"
    >
      $65.00
    </span>
  </div>
</div>
`
